src/Avalonia.Controls/Presenters/PanelContainerGenerator.cs

```diff
@@ -136,12 +136,15 @@ private void ClearItemsControlLogicalChildren()
                 return;
 
             var itemsControl = _presenter.ItemsControl;
+            var generator = itemsControl.ItemContainerGenerator;
             var panel = _presenter.Panel;
 
             foreach (var c in panel.Children)
             {
+                itemsControl.RemoveLogicalChild(c);
+
                 if (!c.IsSet(ItemIsOwnContainerProperty))
```

Comment:
This code was already here, but it looks a bit suspicious. It checks if the `ItemIsOwnContainerProperty` is not set, but it should also check if it's set to `false`, I think?

Reply:
I think it's just mistake made due to inattention. You can look at the local method Remove in OnItemsChanged at this file. When item removed from the list, it is always remove from logical children and IsOwn property used only for clearing.